src/helpers/api.ts

```diff
@@ -1,8 +1,23 @@
-export function fetchMessage(): Promise<string> {
-  // this is just a fake promise resolved in 2 seconds
-  return new Promise(resolve => {
-    setTimeout(() => {
-      resolve('Message from server');
-    }, 2000);
-  });
-}
+const API_URL = 'https://mate-academy.github.io/react_dynamic-list-of-posts/api';
+
+const getAll = <T>(url: string): Promise<T[]> => (
+  fetch(`${API_URL}${url}.json`)
+    .then(response => response.json())
+);
+
+const getPosts = () => getAll<PostFromServer>('/posts');
+const getUsers = () => getAll<UserFromServer>('/users');
+const getComments = () => getAll<CommentFromServer>('/comments');
+
+export const getAppData = async () => {
+  const postsFromServer = await getPosts();
+  const usersFromServer = await getUsers();
+  const commentsFromServer = await getComments();
+
+  return postsFromServer.map((post) => ({
+    ...post,
+    postUser: usersFromServer.find((user: UserFromServer) => post.userId === user.id),
+    postComment: commentsFromServer.filter((comment:
+    CommentFromServer) => post.id === comment.postId),
```

Comment:
you can't break the line like this
```suggestion
    postComment: commentsFromServer
      .filter((comment: CommentFromServer) => post.id === comment.postId),
```